jax/_src/scipy/linalg.py

```diff
@@ -595,3 +611,48 @@ def polar(a, side='right', method='qdwh', eps=None, maxiter=50):
    unitary, posdef, _ = lax_polar.polar(a, side=side, method=method, eps=eps,
                                         maxiter=maxiter)
    return unitary, posdef
+
+@jit
+def _sqrtm_triu(T):
+  diag = jnp.sqrt(jnp.diag(T))
+  n = diag.size
+  U = jnp.diag(diag)
+
+  def i_loop(l, data):
+    j, U = data
+    i = j - 1 - l
+    s = lax.fori_loop(i + 1, j, lambda k, val: val + U[i, k] * U[k, j], 0.0)
+    value = jnp.where(T[i, j] == s, 0.0,
+                      (T[i, j] - s) / (diag[i] + diag[j]))
```

Comment:
Could we add a unit test that explicitly would check this code path? e.g., perhaps a matrix like `[[1, 0, 0], [0, 0,0], [0, 0, 0]]`

Reply:
I added a test for diagonal matrices with zeros